### PR TITLE
Add a ScriptRoot to consolidate global data necessary for multiple passes

### PR DESCRIPTION
**Pull request by jdconrad**
This PR is to get plumbing in for a ScriptRoot class that will consolidate several pieces of state required by potentially multiple passes including PainlessLookup, CompilerSettings, FunctionTable, the root class node, and a synthetic counter. It's possible more may be added to this as we move forward and slowly make the the nodes have less mutable state.

Note a large portion of the code is mechanical since the signature for analyze changed from (SFunction, Locals) -> (ScriptRoot, Locals). The next step will be using the global state from the ScriptRoot and removing unnecessary access to pieces of the global state from other sources. This will be followed up by making the Compiler have the ScriptRoot rather than the SClass node directly.

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/ScriptRoot.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    public CompilerSettings getCompilerSettings() {
+        return compilerSettings;
+    }
+
+    public ScriptClassInfo getScriptClassInfo() {
+        return scriptClassInfo;
+    }
+
+    public SClass getClassRoot() {
```
Comment (rjernst) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/ScriptRoot.java]: maybe getClassNode()?

**Review by rjernst — APPROVED** [review]
LGTM

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/ScriptRoot.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/ScriptRoot.java]: Agreed, better name.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review! Will commit as soon as CI passes.

**Pull request merged**